### PR TITLE
Spotify connection

### PR DESCRIPTION
**Pull request by erxclau**

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -14,25 +15,106 @@
 import functools
 import os
 import json
-import datetime
+from datetime import datetime
+import time
```
Comment (raymondlee00) [inline app/app.py:19]: for spotify expiration

File: app/app.py [hunk 2]
```diff
@@ -14,25 +15,106 @@
...
 
 app = Flask(__name__)
-app.config.from_object(Config)
+# app.config.from_object(Config)
 
 # creates secret key for sessions
 app.secret_key = os.urandom(32)
 
+SPOTIFY_CLIENT_ID = 'b9535e1e2c3741069061954ef75397ab'
+SPOTIFY_CLIENT_SECRET = 'bfe2cc0d745b4047ab805445a0ebb25f'
```
Comment (raymondlee00) [inline app/app.py:37]: can put into another file and read it; gitignore it to keep it secret

File: app/app.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 '''
 subtleAsianCoders - Yaru Luo, Matthew Chan, Eric "Morty" Lau, ray. lee.
 SoftDev1 pd1
-P02 -- RYthEM
+P02 -- The End
+RYthEM
```
Comment (MattChann) [inline app/app.py:5]: Duende

File: app/templates/base.html [hunk 4]
```diff
@@ -1,50 +1,70 @@
...
 
 <!--
-    Team subteAsianCoders -- Yaru Luo (PM), Matthew Chan, Eric Lau, Raymond Lee
-    SoftDev1 pd1
-    P02 -- The End
-    2020-01-07
+subtleAsianCoders - Yaru Luo, Matthew Chan, Eric "Morty" Lau, ray. lee.
+SoftDev1 pd1
+P02 -- The End
+RYthEM
```
Comment (MattChann) [inline app/templates/base.html:7]: Duende

File: app/app.py (hunk 3 above)
Comment (MattChann) [inline app/app.py:5]: Duende

File: app/templates/base.html [hunk 5]
```diff
@@ -1,50 +1,70 @@
...
+            integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo"
+            crossorigin="anonymous"></script>
+        <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
+            integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6"
+            crossorigin="anonymous"></script>
 
 
         <title>
-            {% block title %}BASE{% endblock %} :: Duende
+            {% block title %}{% endblock %} :: RYthEM
```
Comment (MattChann) [inline app/templates/base.html:36]: Duende

File: app/templates/base.html [hunk 6]
```diff
@@ -1,50 +1,70 @@
...
         </title>
     </head>
+
     <body>
-        <nav class="navbar navbar-expand-sm bg-dark navbar-dark justify-content-left fixed-top" style="margin-bottom:50px;">
-            <a class="navbar-brand" href="{{ url_for("home") }}">Home</a>
+        {% block nav %}
```
Comment (MattChann) [inline app/templates/base.html:41]: Remove this

File: app/templates/base.html [hunk 7]
```diff
@@ -1,50 +1,70 @@
...
+                    <a class="nav-link" href="/hearted_songs">My Hearted Songs</a>
+                </li>
+            </ul>
+            <ul class="navbar-nav ml-auto">
+                <li class="nav-item navbar-right">
+                    <a class="nav-link text-success" href="/spotify_connect">Connect to Spotify</a>
                 </li>
             </ul>
         </nav>
+        {% endblock %}
```
Comment (MattChann) [inline app/templates/base.html:61]: Remove, no need

File: app/templates/base.html [hunk 8]
```diff
@@ -1,50 +1,70 @@
...
 
+        <div class="container-fluid">
```
Comment (MattChann) [inline app/templates/base.html:63]: Make another block to change background image/color that is edge-to-edge

File: app/templates/home.html [hunk 9]
```diff
@@ -0,0 +1,35 @@
+{% extends "base.html" %}
+
+{% block title %}
+Home
+{% endblock %}
+
+{% block nav %}
+{{ super() }}
```
Comment (MattChann) [inline app/templates/home.html:8]: Don't need super(), just dont use block nav

File: app/templates/index.html [hunk 10]
```diff
@@ -1,9 +1,5 @@
...
-  </head>
-  <body>
-    Hello World
-  </body>
-</html>
+{% extends "base.html" %}
+
+{% block body %}
+{{ data }}
+{% endblock %}
```
Comment (MattChann) [inline app/templates/index.html:5]: Delete

File: doc/devlog.txt [hunk 11]
```diff
@@ -1,7 +1,7 @@
-subtleAsianCoders - Yaru Luo, Eric "Morty" Lau, Raymond Lee, Matthew Chan
+subtleAsianCoders - Yaru Luo, Matthew Chan, Eric "Morty" Lau, ray. lee.
 SoftDev1 pd1
 P02 -- The End
-DEVLOG
+RYthEM DEVLOG
```
Comment (MattChann) [inline doc/devlog.txt:4]: Duende

**Review by MattChann — CHANGES_REQUESTED** [review]

**Review by raymondlee00 — APPROVED** [review]

**Pull request merged**